packages/jellyfish-testing/src/icxorderbook.ts

```diff
@@ -0,0 +1,314 @@
+    this.poolOwner = await this.container.call('getnewaddress', ['', 'legacy'])
+    await accountToAccount(this.container, this.symbolBTC, 1, { from: this.accountBTC, to: this.accountDFI })
+
+    const poolPairMetadata = {
+      tokenA: this.idBTC,
+      tokenB: this.idDFI,
+      commission: 1,
+      status: true,
+      ownerAddress: this.poolOwner,
+      pairSymbol: 'BTC-DFI'
+    }
+    await this.container.call('createpoolpair', [poolPairMetadata, []])
+    await this.container.generate(1)
+
+    const pool = await this.container.call('getpoolpair', ['BTC-DFI'])
+    const combToken = await this.container.call('gettoken', ['BTC-DFI'])
+    const idDFIBTC = Object.keys(combToken)[0]
+    expect(pool[idDFIBTC].idTokenA).toBe(this.idBTC)
+    expect(pool[idDFIBTC].idTokenB).toBe(this.idDFI)
+  }
+
+  async addLiquidityToBTCDFIPool (amountInBTC: number, amountInDFI: number): Promise<void> {
+    const poolLiquidityMetadata: { [key: string]: string [] } = {}
+    poolLiquidityMetadata[this.accountDFI] = [`${amountInBTC}@${this.symbolBTC}`, `${amountInDFI}@${this.symbolDFI}`]
+
+    await this.container.call('addpoolliquidity', [poolLiquidityMetadata, this.accountDFI, []])
+    await this.container.generate(1)
+    this.DEX_DFI_PER_BTC_RATE = amountInDFI / amountInBTC
+  }
+
+  async setTakerFee (fee: number): Promise<void> {
+    await this.container.call('setgov', [{ ICX_TAKERFEE_PER_BTC: fee }])
+    await this.container.generate(1)
+    const result: any = await this.container.call('getgov', ['ICX_TAKERFEE_PER_BTC'])
+    expect(result.ICX_TAKERFEE_PER_BTC as number).toStrictEqual(fee)
+    this.ICX_TAKERFEE_PER_BTC = result.ICX_TAKERFEE_PER_BTC as number
+  }
+
+  async closeAllOpenOffers (): Promise<void> {
+    const orders = await this.container.call('icx_listorders', [])
+    for (const orderTx of Object.keys(orders).splice(1)) {
+      const offers = await this.container.call('icx_listorders', [{ orderTx: orderTx }])
+      for (const offerTx of Object.keys(offers).splice(1)) {
+        if (offers[offerTx].status === ICXOrderStatus.OPEN) {
+          await this.container.call('icx_closeoffer', [offerTx])
+        }
+      }
+    }
+    await this.container.generate(1)
+  }
+
+  // creates DFI sell order
+  async createDFISellOrder ({
+    chainTo = 'BTC',
+    ownerAddress = this.accountDFI,
+    receivePubkey,
+    amountFrom,
+    orderPrice
+  }: CreateDFISellOrderData): Promise<{order: icxorderbook.ICXOrder, createOrderTxId: string}> {
+    // create order - maker
+    const order = {
+      tokenFrom: this.idDFI,
+      chainTo,
+      ownerAddress,
+      receivePubkey,
+      amountFrom,
+      orderPrice
+    }
+
+    const { txid: createOrderTxId } = await this.rpc.icxorderbook.createOrder(order)
+    await this.container.generate(1)
+
+    const ordersAfterCreateOrder = await this.rpc.icxorderbook.listOrders()
+    expect(ordersAfterCreateOrder[createOrderTxId].status).toStrictEqual(ICXOrderStatus.OPEN)
```

Comment:
maybe remove this expect and line above since now we have a separate test file for these functions. and add the test there if not already.